org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/QuickAssistProcessor.java

```diff
@@ -4635,4 +4640,97 @@ private boolean getSplitSwitchLabelProposal(IInvocationContext context, ASTNode
 
 		return true;
 	}
+	private boolean getConvertFieldNamingConventionProposal(IInvocationContext context, ASTNode node, Collection<ICommandAccess> resultingCollections) {
+		if (!(node instanceof SimpleName)) {
+			return false;
+		}
+		SimpleName name= (SimpleName) node;
+		String selectedField = name.toString();
+		int offset = node.getStartPosition();
+		int length = node.getLength();
+		if (name.getAST().apiLevel() >= ASTHelper.JLS10 && name.isVar()) {
+			return false;
+		}
+
+		IEditorPart editor= ((AssistContext) context).getEditor();
+		if (!(editor instanceof JavaEditor))
+			return false;
+
+        @SuppressWarnings("unused")
+		List<String> fieldList = new ArrayList<String>();
+		CompilationUnit astRoot= context.getASTRoot();
+        TypeDeclaration typeDec = (TypeDeclaration) astRoot.types().get(0);
```

Comment:
This needs to be fixed.  The logic is wrong.  You don't always want the first type in the file.  This could be a field in a local or static internal class.  Looking for a name could get you a duplicate name in another class.

What you want to do is to resolve the binding of the name.  If it fails, then stop.  Otherwise, you can check if you have an IVariableBinding which then you can query: isFieldBinding().  If not a field, return.

You can then use ASTNodes.findDeclaration(IVariableBinding) to get the VariableDeclarationFragment of a FieldDeclaration (you can use getParent() if FieldDeclaration is needed).  From there, you can check modifiers.